### PR TITLE
prov/rxm: Extend the rxm to support the underlying providers with FI_CONTEXT mode set

### PR DESCRIPTION
**Pull request by gladkovdmitry17**
Signed-off-by: Dmitry Gladkov <dmitry.gladkov@intel.com>

### CODE REVIEW

**Comment by gladkovdmitry17** [comment]
@a-ilango @shefty 
These are changes for the rxm provider that are separated from the PR #3269

File: prov/rxm/src/rxm.h [hunk 1]
```diff
@@ -86,6 +86,11 @@
 	       " (fi_addr: 0x%" PRIx64 " tag: 0x%" PRIx64 ")\n",\
 	       addr, tag)
 
+#define RXM_GET_PROTO_STATE(comp)			\
+	((enum rxm_proto_state *)			\
+	 (((unsigned char *)comp->op_context +		\
+		offsetof(struct rxm_buf, state))))
```
Comment (shefty) [inline prov/rxm/src/rxm.h]: Can we update this to return the state, rather than a pointer to the state?

Comment (gladkovdmitry17) [reply prov/rxm/src/rxm.h]: Sure, no problem.

**Comment by gladkovdmitry17** [comment]
@a-ilango @shefty I've updated `RXM_GET_PROTO_STATE` to return the state, rather than a pointer to the state

**Comment by shefty** [comment]
See build error:
(ClCompile target) -> 
  c:\projects\libfabric\prov\rxm\src\rxm_cq.c(498): error C2100: illegal indirection
There's a spot where the state needs to be updated.  I missed this when I asked to update the function. Maybe add a SET_STATE macro?

**Comment by gladkovdmitry17** [comment]
@shefty YEs, it's nice idea. Will do that

**Comment by gladkovdmitry17** [comment]
@shefty @a-ilango I've updated to cover the remarks. Is there are possibility to get this merged? Thank you.

**Pull request merged**